classes/submission/SubmissionDAO.inc.php

```diff
@@ -111,10 +112,104 @@ function _fromRow($row) {
 		$submission->setLanguage($row['language']);
 		$submission->setCommentsToEditor($row['comments_to_ed']);
 
-		$this->getDataObjectSettings('submission_settings', 'submission_id', $submission->getId(), $submission);
+		$this->getSubmissionSettings($submission, $submissionRevision);
 
 		return $submission;
 	}
+	
+	/**
+	 * Get contents of the submission_settings table
+	 * @param $submission Submission object
+	 * @param $submissionRevision int optional
+	 */
+	function getSubmissionSettings($submission, $submissionRevision = null) {
```

Comment:
(I wonder if we're going to have to scale out revisions beyond just the submissions table, e.g. to include author names, galley information, etc. In that case it might be worth making this an optional feature of DAO::getDataObjectSettings.)